hw3/hw3-v1tal4ik/src/v1tal4ik.js

```diff
@@ -0,0 +1,161 @@
+// eslint-disable-next-line max-classes-per-file
+import validateObj from './helperFunc';
+import * as objUsers from './people.data';
+
+// 1. write a constructor function `PersonFunc` whith methods : getFullName() getAge() getFullAddress();
+const PersonFunc = function (obj) {
+  const { isObjHasAllProperty, isObjPropertyTypeRight, msg } = validateObj(obj);
+  if (!isObjHasAllProperty || !isObjPropertyTypeRight) {
+    return { msg };
+  }
+  this.person = obj;
+};
+
+
+PersonFunc.prototype.getFullName = function () { return `${this.person.firstName} ${this.person.lastName}`; };
+PersonFunc.prototype.getFullAddress = function () { return `${this.person.address.country}, ${this.person.address.city}, ${this.person.address.street}, ${this.person.address.house}/${this.person.address.apartment}`; };
```

Comment:
aprtment is not required field. It's absence shouldn't throw an error. What will happen if in the given object apartment field is empty?

Reply:
You allowed to create handle error on my own and I decided all of the properties required.
Yes, I need to add check if field is empty.